### PR TITLE
feat(cli): v2 configure cmd undestands subaccounts

### PR DESCRIPTION
**Pull request by afiune**
This change also adds a global flag `--subaccount` to switch between
accounts. (ALLY-378)

**User Story**
As a Lacework CLI User, I would like to avoid introducing my account and
sub-account information, So that I don't do any mistakes and my CLI
configuration is faster.

**Description**
The Lacework CLI needs to understand the new JSON fields that users can
download from their accounts:
```json
{
  "keyId": "MINIALLY_138DB0DAA9084BD93E50F587A58DEA572E543A5C8C20006",
  "secret": "_f28a1c84014f34b948422395fb0691f3",
  "subAccount": "mini-ally",
  "account": "customerdemo.lacework.net"
}
```
The two new fields, `account`, and `subAccount`, will be stored inside
the configuration file stored at `~/.lacework.toml` and marked as
`version 2`
```toml
[default]
  account = "CORP"
  subaccount = "CORP-PROD"
  api_key = "CORP_A3C8910D0BDCDBD7AFD4355A1C5284104AAA2AE5953938C"
  api_secret = "_1234abca752e65f94651bcb6e6c798f1"
  version = 2
```

This change is backward compatible with the previous JSON format for
API keys as well as previous profiles inside the configuration file.

**Acceptance Criteria**
After completing this task, new Lacework CLI users that download the new
JSON format of an API key from the UI will be able to configure the CLI
without any human interaction (since all necessary fields are already
inside the JSON file).

Additionally, old profiles configured should continue to work.

JIRA: ALLY-374

Signed-off-by: Salim Afiune Maya <afiune@lacework.net>

### CODE REVIEW

File: api/user_profile.go [hunk 1]
```diff
@@ -30,18 +35,53 @@ func (svc *UserProfileService) Get() (response UserProfileResponse, err error) {
...
+	Username   string    `json:"username"`
+	OrgAccount bool      `json:"orgAccount"`
+	URL        string    `json:"url"`
+	OrgAdmin   bool      `json:"orgAdmin"`
+	OrgUser    bool      `json:"orgUser"`
+	Accounts   []Account `json:"accounts"`
+}
+
+func (p *UserProfile) OrgAccountName() string {
+	// TODO @afiune should we handle other datacenters?
```
Comment (afiune) [inline api/user_profile.go:51]: I am trying to get access to an account deployed in our Frankfurt datacenter 

tenor-54643987

File: api/user_profile.go [hunk 2]
```diff
@@ -30,18 +35,53 @@ func (svc *UserProfileService) Get() (response UserProfileResponse, err error) {
...
+	rx, err := regexp.Compile(`\.lacework\.net.*`)
+	if err == nil {
+		if split := rx.Split(p.URL, -1); len(split) != 0 {
+			return strings.ToLower(split[0])
+		}
+	}
+
+	return p.URL
+}
```
Comment (dmurray-lacework) [inline api/user_profile.go:60]: Are there any tests for this regex? Ideally that tests US and EU.

File: api/client.go [hunk 3]
```diff
@@ -123,11 +123,23 @@ func NewClient(account string, opts ...Option) (*Client, error) {
 	return c, nil
 }
 
-// WithSubaccount sets a subaccount
+// RemoveSubaccount removes the subaccount from the API client
+func (c *Client) RemoveSubaccount() {
```
Comment (dmurray-lacework) [inline api/client.go:127]: What situation would the subaccount need to be removed?

^ Not a comment on the usefulness of this function, genuinely not sure of the use case to remove the subaccount.

File: api/user_profile.go [hunk 4]
```diff
@@ -30,18 +35,53 @@ func (svc *UserProfileService) Get() (response UserProfileResponse, err error) {
...
+	names := make([]string, 0)
+	orgAccountName := p.OrgAccountName()
+	for _, acc := range p.Accounts {
+		accName := strings.ToLower(acc.AccountName)
+		if accName == orgAccountName {
+			continue
+		}
+		if acc.Enabled() {
+			names = append(names, accName)
+		}
```
Comment (dmurray-lacework) [inline api/user_profile.go:72]: Can this logic be condensed? Not sure which is more readable to be fair. Ignore if you don't like it (or my logic is incorrect)
```suggestion
		if accName != orgAccountName && acc.Enabled() {
			names = append(names, accName)
		}
```

**Review by dmurray-lacework — APPROVED** [review]

File: api/client.go (hunk 3 above)
Comment (afiune) [reply api/client.go:127]: To access the primary account, instead of the sub-account

File: api/user_profile.go (hunk 4 above)
Comment (afiune) [reply api/user_profile.go:72]: I'm fine with either one of them. 🙌🏽

File: api/user_profile.go (hunk 1 above)
Comment (afiune) [reply api/user_profile.go]: tenor-168906724

File: api/user_profile.go (hunk 2 above)
Comment (afiune) [reply api/user_profile.go]: Now they are 🙌🏽

tenor-67817893

File: api/client.go (hunk 3 above)
Comment (afiune) [reply api/client.go:127]: Just realized that this was part of the next change for the migration. I'll remove it from here. 👍🏽

**Pull request merged**